messaging/src/main/protos/Model.proto

```diff
@@ -0,0 +1,155 @@
+  enum DeliveryStatus {
+    SENDING = 0; // message is currently pending send but hasn't successfully finished yet
+    PARTIALLY_SENT = 1; // message has been successfully sent to some of the recipients, will retry other recipients
+    COMPLETELY_SENT = 2; // message has been successfully sent to all recipients
+    PARTIALLY_FAILED = 3; // message failed to send to a subset of recipients, will not retry
+    COMPLETELY_FAILED = 4; // message failed to send to any recipients, will not retry
+  }
+
+  ContactId contactId = 1; // the ContactId that identifies the conversation to which this message belongs (either a direct conversation with a person, or messages in a group)
+  string senderId = 2; // the id of the person who sent this message (will match the conversation.id for direct conversations)
+  string id = 3; // the id of the message (same as on Message), encoded in base32
+  int64 ts = 4; // The unix timestamp in milliseconds for when the message was sent (for OUT) or received (for IN)
+  string replyToSenderId = 5; // if this message is a reply to another message, this contains the ID of the sender of that message
+  string replyToId = 6; // if this message is a reply to another message, this contains the ID of that message
+  string text = 7;
+  int32 disappearAfterSeconds = 8;
+  map<int32, StoredAttachment> attachments = 9; // attachments keyed to unique IDs for each attachment within this message (excluding thumbnails)
+  map<int32, int32> thumbnails = 15; // maps the id of a thumbnail attachment to the attachment for which it is a thumbnail
+  MessageDirection direction = 10; // direction in which the message is going/came
+  map<string, Reaction> reactions = 11; // reactions by the id of the reacter
+  DeliveryStatus status = 12; // for outgoing messages, the status of its sending and delivery
+  int64 firstViewedAt = 13; // the unix timestamp in milliseconds of when this message was first viewed on the current device
+  int64 disappearAt = 14; // the unix timestamp in milliseconds of when this message will automatically disappear
+}
+
+// A reaction to a message
+message Reaction {
+  bytes reactingToSenderId = 1; // the id of the sender of the message to which this is reacting
+  bytes reactingToMessageId = 2; // the id of the message to which this is reacting
+  string emoticon = 3; // a 2 byte emoticon representing the reaction
+}
+
+// An update to disappearing message settings
+message DisappearSettings {
+  int32 messagesDisappearAfterSeconds = 1; // the number of seconds after which new messages will automatically disappear. <=0 means they won't disappear.
+}
+
+// An envelope for messages to be transferred via the message broker
+message TransferMessage {
+  oneof content {
+    bytes message = 1; // serialized Message
+    bytes reaction = 2; // serialized Reaction
+    bytes deleteMessageId = 3; // the id of a message from this Sender to delete
+    bytes disappearSettings = 4; // serialized DisappearSettings
```

Comment:
Because they're already serialized on the OutboundMessage and there's no point in deserializing them before sending them on. The reason they're already serialized on the OutboundMessage is that the OutboundMessage is essentially a fancy work-queue entry that can potentially be read and persisted to the database multiple times, and if we kept these things typed they'd incur the serialization/deserialization overhead each time. The stuff on the OutboundMessage that needs to be read/modified for the work queue logic is fielded, but this stuff here is just payload that doesn't change.